lib/kubernetes-deploy/kubernetes_resource/service.rb

```diff
@@ -11,16 +11,19 @@ def sync
     end
 
     def status
-      if @num_pods_selected.blank?
+      if !requires_endpoints?
+        "Doesn't require any endpoint"
+      elsif @num_pods_selected.blank?
         "Failed to count related pods"
       elsif selects_some_pods?
         "Selects at least 1 pod"
       else
-        "Selects 0 pods"
+        "Unknown"
       end
     end
 
     def deploy_succeeded?
+      return exists? unless requires_endpoints?
```

Comment:
The service must be found no matter if the service is external or not. Line 26 says that if the service is external then it is deployed successfully. It may not be true.

Reply:
> The service must be found no matter if the service is external or not.

Do you mean L28 should be `exists? && (exposes_zero_replica_deployment? || selects_some_pods?)`? It's true that the pod selection doesn't inherently prove the server-side service resource was created, though that problem would predate this PR. Come to think of it, this could now just be `exists? && selects_some_pods?` since `exposes_zero_replica_deployment?` is already covered by the `requires_endpoints?` check.

>  Line 26 says that if the service is external then it is deployed successfully. It may not be true.

What do you mean? If the service is external, afaik there's no good way to check whether it is in fact available, so server-side existence is the best indicator we have.

Reply:
Please ignore my comment about line 26. I was confused. The code is right.

Reply:
👍